include/drivers/lora.h

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2019 Manivannan Sadhasivam
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_LORA_H_
+#define ZEPHYR_INCLUDE_DRIVERS_LORA_H_
+
+/**
+ * @file
+ * @brief Public LoRa APIs
+ */
+
+#include <zephyr/types.h>
+#include <device.h>
+
+enum lora_signal_bandwidth {
+	BW_125_KHZ = 0,
+	BW_250_KHZ,
+	BW_500_KHZ,
+};
+
+enum lora_spreading_factor {
+	SF_6 = 6,
+	SF_7,
+	SF_8,
+	SF_9,
+	SF_10,
+	SF_11,
+	SF_12,
+};
+
+enum lora_coding_rate {
+	CR_4_5 = 1,
+	CR_4_6 = 2,
+	CR_4_7 = 3,
+	CR_4_8 = 4,
+};
+
+struct lora_modem_config {
+	u32_t frequency;
+	int bandwidth;
+	int spreading_factor;
+	int preamble_len;
+	int coding_rate;
+	int tx_power;
+	bool tx;
```

Comment:
to keep proper struct alignement after the type changes I noted (unless you have good reasons to keep what you have), you could move attributes so premble_len goes before tx_power. etc...

Reply:
ok